backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/AdminController.java

```diff
@@ -1,23 +1,116 @@
 package ba.unsa.etf.si.bbqms.ws.controllers;
 
 import ba.unsa.etf.si.bbqms.admin_service.api.AdminService;
+import ba.unsa.etf.si.bbqms.auth_service.api.AuthService;
+import ba.unsa.etf.si.bbqms.domain.User;
+import ba.unsa.etf.si.bbqms.ws.models.ErrorMessage;
+import ba.unsa.etf.si.bbqms.ws.models.ErrorResponseDto;
+import ba.unsa.etf.si.bbqms.ws.models.SimpleMessageDto;
+import ba.unsa.etf.si.bbqms.ws.models.UserDto;
+import jakarta.persistence.EntityNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
 
 @RestController
 @RequestMapping("/api/v1/admin")
 public class AdminController {
+    private static final Logger logger = LoggerFactory.getLogger(AuthController.class);
     private final AdminService adminService;
+    private final AuthService authService;
 
-    public AdminController(final AdminService adminService) {
+    public AdminController(final AdminService adminService, final AuthService authService) {
         this.adminService = adminService;
+        this.authService = authService;
+    }
+
+    @GetMapping("/{code}")
+    @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
+    public ResponseEntity getAdmins(@PathVariable(name = "code") final String tenantCode) {
+        final User currentUser = this.authService.getAuthenticatedUser();
+
+        if (!currentUser.getTenant().getCode().equals(tenantCode)) {
+            logger.warn("Super admin does not belong to the specified tenant");
+            return ResponseEntity.badRequest().body(new ErrorMessage("Couldn't get admins"));
+        }
+        try {
+            final List<User> admins = this.adminService.findAdminsByCode(tenantCode);
+            List<UserDto> adminDtos = new ArrayList<>();
+
+            for (User admin : admins) {
+                adminDtos.add(UserDto.fromEntity(admin));
+            }
+            return ResponseEntity.ok(adminDtos);
```

Comment:
This can be shortened:

```
return ResponseEntity.ok().body(
                this.adminService.findAdminsByCode(tenantCode).stream()
                        .map(UserDto::fromEntity)
                        .collect(Collectors.toList())
);
```